test/limitOrderBook/limitOrderBook.fillOrder.test.ts

```diff
@@ -0,0 +1,575 @@
+    let pool2: UniswapV3Pool
+    let mockedBaseAggregator: FakeContract<TestAggregatorV3>
+    let mockedBaseAggregator2: FakeContract<TestAggregatorV3>
+    let collateralDecimals: number
+    let rewardToken: TestERC20
+    let limitOrderFeeVault: LimitOrderFeeVault
+
+    const fakeSignature = "0x0000000000000000000000000000000000000000000000000000000000000000"
+
+    beforeEach(async () => {
+        fixture = await loadFixture(createLimitOrderFixture())
+        limitOrderBook = fixture.limitOrderBook
+        clearingHouse = fixture.clearingHouse as TestClearingHouse
+        delegateApproval = fixture.delegateApproval
+
+        orderBook = fixture.orderBook
+        accountBalance = fixture.accountBalance as TestAccountBalance
+        clearingHouseConfig = fixture.clearingHouseConfig
+        vault = fixture.vault
+        exchange = fixture.exchange
+        marketRegistry = fixture.marketRegistry
+        collateral = fixture.USDC
+        baseToken = fixture.baseToken
+        baseToken2 = fixture.baseToken2
+        quoteToken = fixture.quoteToken
+        mockedBaseAggregator = fixture.mockedBaseAggregator
+        mockedBaseAggregator2 = fixture.mockedBaseAggregator2
+        pool = fixture.pool
+        pool2 = fixture.pool2
+        collateralDecimals = await collateral.decimals()
+        limitOrderFeeVault = fixture.limitOrderFeeVault
+        rewardToken = fixture.rewardToken
+
+        const pool1LowerTick: number = priceToTick(2000, await pool.tickSpacing())
+        const pool1UpperTick: number = priceToTick(4000, await pool.tickSpacing())
+        delegateApproval = fixture.delegateApproval
+        limitOrderBook = fixture.limitOrderBook
+
+        // ETH
+        await initAndAddPool(
+            fixture,
+            pool,
+            baseToken.address,
+            encodePriceSqrt("2960", "1"),
+            10000, // 1%
+            // set maxTickCrossed as maximum tick range of pool by default, that means there is no over price when swap
+            getMaxTickRange(),
+        )
+        await syncIndexToMarketPrice(mockedBaseAggregator, pool)
+
+        // prepare collateral for maker
+        await mintAndDeposit(fixture, maker, 1_000_000_000_000)
+
+        // maker add liquidity
+        await clearingHouse.connect(maker).addLiquidity({
+            baseToken: baseToken.address,
+            base: parseEther("3000"),
+            quote: parseEther("10000000"),
+            lowerTick: pool1LowerTick,
+            upperTick: pool1UpperTick,
+            minBase: 0,
+            minQuote: 0,
+            useTakerBalance: false,
+            deadline: ethers.constants.MaxUint256,
+        })
+
+        // prepare collateral for taker
+        await mintAndDeposit(fixture, trader, 1000)
+
+        await delegateApproval.connect(trader).approve([
+            {
+                delegate: limitOrderBook.address,
+                action: fixture.clearingHouseOpenPositionAction,
+            },
+        ])
+    })
+
```

Comment:
we need one more case to specify the spec. what will happen if we short 0.1 position in this case? should be success or not?